core/src/main/kotlin/org/evomaster/core/problem/rest/service/SmartSamplingController.kt

```diff
@@ -0,0 +1,345 @@
+package org.evomaster.core.problem.rest.service
+
+import com.google.inject.Inject
+import org.evomaster.core.EMConfig
+import org.evomaster.core.problem.rest.resource.RestResourceNode
+import org.evomaster.core.problem.rest.resource.SamplerSpecification
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.SearchTimeController
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * dynamically determinate resource-based sample method
+ */
+class SmartSamplingController {
+
+    @Inject
+    private lateinit var time : SearchTimeController
+
+    @Inject
+    private lateinit var randomness: Randomness
+
+    @Inject
+    private lateinit var config : EMConfig
+
+    @Inject
+    private lateinit var rm : ResourceManageService
+
+    companion object {
+        private val log: Logger = LoggerFactory.getLogger(SmartSamplingController::class.java)
+        private const val CONARCHIVE_THRESHOLD = 0.2
+        private const val TB_THRESHOLD = 0.5
+
+    }
+
+    fun initialize(){
+        initApplicableStrategies()
+        initProbability()
+        validateProbability()
+    }
+
+    private fun initApplicableStrategies(){
+        ResourceSamplingMethod.S1iR.applicable = true //mutableMap.values.filter { r -> r.hasIndependentAction }.isNotEmpty()
```

Comment:
I am bit skeptical about this `ResourceSamplingMethod` enum. Unless I misunderstanding, it would look like a global class with 4 instances with mutable state. If that is indeed the case, it would be equivalent to a static state, which we should avoid in EvoMaster

Reply:
thanks, fixed.